python/cuda_parallel/cuda/parallel/experimental/iterators.py

```diff
@@ -31,4 +32,4 @@ def CountingIterator(offset):
 
 def TransformIterator(op, it):
     """Python facade (similar to built-in map) mimicking a C++ Random Access TransformIterator."""
-    return _iterators.TransformIterator(op, it)
+    return _iterators.make_transform_iterator(it, op)
```

Comment:
I should have left a comment here clarifying why I reversed the argument order:

1. The order `it, op` is the same as in [`thrust::transform_iterator`](https://nvidia.github.io/cccl/thrust/api/classthrust_1_1transform__iterator.html#_CPPv4N6thrust18transform_iterator18transform_iteratorERK8Iterator22AdaptableUnaryFunction). As discussed in [this thread](https://github.com/NVIDIA/cccl/pull/2788#discussion_r1873235103), we should build an API as similar to the underlying C++ layer as possible - and in the future, a more Pythonic one on top of that.

2. Since we're no longer using the name `map`, it's less likely to cause confusion. I absolutely agree that a function called `map(it, op)` would trip folks up.